File: shared-backend/models/CarBrand.js
const mongoose = require('mongoose');

const carBrandSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true,
    trim: true
  },
  logo: {
    type: String,
    default: null
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

// Index for efficient search
carBrandSchema.index({ name: 1, isActive: 1 });

module.exports = mongoose.model('CarBrand', carBrandSchema);

File: clutch-app-android/app/src/main/java/com/clutch/app/data/model/CarBrand.kt
package com.clutch.app.data.model

import com.google.gson.annotations.SerializedName

data class CarBrand(
    @SerializedName("_id")
    val id: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("logo")
    val logo: String? = null,
    @SerializedName("isActive")
    val isActive: Boolean = true
)
